components/datetime/src/neo.rs

```diff
@@ -365,6 +365,90 @@ where
     }
 }
 
+impl<C: CldrCalendar, FSet: DateTimeNamesMarker> FixedCalendarDateTimeFormatter<C, FSet> {
+    /// Formats a datetime without enforcing either the field set or the calendar.
+    ///
+    /// This function is useful when the caller knows something about the field set that the
+    /// type system is unaware of. For example, if the formatter is represented with a
+    /// [dynamic field set](crate::fieldsets::enums), the caller may be able to provide a
+    /// narrower type for formatting.
+    ///
+    /// ❗ The caller must ensure that:
+    ///
+    /// 1. The calendar of the input matches the calendar of the formatter
```

Comment:
do we need to be unchecked in both dimensions?

Reply:
No; I could make it unchecked in only one dimension. But I don't see a good reason to do that. If there's a function named "unchecked", let's just relax both dimensions.